src/views/Generator/Generator.hooks.ts

```diff
@@ -89,3 +90,39 @@ export function useIsGeneratorDirty(fileName: string) {
     JSON.stringify(generatorStateData) !== JSON.stringify(generatorFileData)
   )
 }
+
+export function useScriptExport(generatorFileName: string) {
+  const showToast = useToast()
+  const setScriptName = useGeneratorStore((store) => store.setScriptName)
+  const { mutateAsync: updateGeneratorFile } =
+    useUpdateValueInGeneratorFile(generatorFileName)
+
+  return useCallback(
+    async (scriptName: string) => {
+      setScriptName(scriptName)
+
+      try {
+        await exportScript(scriptName)
+      } catch (error) {
+        log.error(error)
+
+        showToast({
+          title: 'Failed to export script',
+          status: 'error',
+        })
+      }
+
+      try {
+        await updateGeneratorFile({ key: 'scriptName', value: scriptName })
+      } catch (error) {
+        log.error(error)
+
+        showToast({
+          title: 'Failed to update script name',
+          status: 'error',
+        })
+      }
```

Comment:
This is the generator specific logic that I've lifted out of `ExportScriptDialog`.

We previously didn't await `exportScript` so errors exporting the script would, from what I can tell, just get swallowed. 

I considered in which order to do things: update the script name in the generator first or export the script first.

In the end I went for script first because I think that blocking the user from exporting a script because we couldn't update the script name in the generator would be frustrating (though technically we could do them in parallel, I suppose 🤔 ).